Third-party-libraries.md

```diff
@@ -143,3 +158,98 @@ We download and install the Redis CLI and Elasticsearch development server. We p
 ### `install_backend_python_libs.py`
 
 This script installs the Python dependencies we need in production to `third_party/python_libs`. We define these dependencies using `requirements.in`, which lists the libraries we depend on directly. Then `install_backend_python_libs.py` runs `scripts.regenerate_requirements` to generate `requirements.txt`, which lists those direct dependencies, plus all the packages that our direct dependencies need, and so on. Both `requirements.in` and `requirements.txt` specify versions, so `requirements.txt` is analogous to `yarn.lock` in that it pins all the versions of all the Python packages we use in production.
+
+## Upgrade dependencies
+
+### Upgrade backend libraries
+
+Before upgrading any backend dependencies, you should check for any breaking changes between the currently installed version and the new version you want to upgrade the library to. You should:
+
+* Check the library's changelog for breaking changes.
```

Comment:
It seems to me that you mention checking of breaking changes twice.

Reply:
The first time is in the section for backend tests. The second time is in the section for frontend tests in `manifest.json`. I did this because there's a separate procedure for npm libraries

Reply:
No, I mean here `you should check for any breaking changes between the currently installed version and the new version you want to upgrade the library to` and then `Check the library's changelog for breaking changes.` seems a bit redundant to me.

Reply:
Oh, I understand now. The bullet points are explaining how to check for breaking changes. I want developers to check for breaking changes two ways:

1. Read the library's changelog
2. After upgrading the library, test that nothing broke

Does that make sense? The first of the phrases you mention is me saying that you need to check for breaking changes. The second is listing one of the ways to perform that check.

Reply:
Ah, okay, makes sense.